### ISSUE TITLE
skip tokenization for tokenized corpora

### ISSUE THREAD

**Issue by dominikacecylia** [modernmt/modernmt#581]
Is it possible to skip the tokenization step in training? I have tokenized corpora but could not find the skip option in training parameters.

**Comment by mzeidhassan** [comment]
I am not sure if there is a way to do it through the command line, but if you have MMT built from source, I think you can comment this line

`<processor>eu.modernmt.processing.tokenizer.Tokenizer</processor>`

in 

"modernmt/src/textprocessing/src/main/resources/eu/modernmt/processing/preprocessor-default.xml"

and rebuild MMT again, then run your training. 

Good luck!

**Comment by davidecaroselli** [comment]
Hi @dominikacecylia 

ModernMT pre-processing pipeline (including tokenization) is a mandatory step in the process. Unfortunately, the hint of @mzeidhassan won't work because the tokenizer is needed to identify words. If the tokenizer is not run, ModernMT will simply find a sentence without `Word`s.

However, what you can do is follow the instructions of @mzeidhassan but, instead of removing the tokenizer, you can write your own "WhitespaceTokenizer" implementation, which only relies on whitespaces to identify tokens.

A simpler solution could be to just run a de-tokenizer on your training corpus. This might be the easiest fix if you have such a de-tokenizer script.

**Comment by mzeidhassan** [comment]
Thanks @davidecaroselli for your feedback and support.

I have one question that is related somehow:

Is the [current instructions](https://github.com/modernmt/modernmt/wiki/FAQs#q-during-engine-creation-the-pre-processing-step-throws-a-corpus-not-found-error-what-can-i-do) on how to disable cleaning still valid?

**Comment by EtienneAb3d** [comment]
Remark: it seems wiki pages have a problem. I see HTML code in place of the rendered viewing. A github bug?

I'm always using the `--skip-cleaning` option, and it seems to be properly recognized.

**Comment by dominikacecylia** [comment]
@EtienneAb3d the --skip-cleaning option also works for me.

@davidecaroselli thanks for the tip, I detokenized my data as it seemed the easiest tip.
Also about tokenization, does it work for Czech or Slovak? As it failed when I tried my corpora, but maybe that was just the file issue not the tokenizer. I use cs or sk as the language abbreviations.

**Comment by mzeidhassan** [comment]
Thanks @EtienneAb3d. I appreciate it. 
Yes, it seems there is something wrong with the click-through link, that's why I copied the main page link there.

**Comment by davidecaroselli** [comment]
> Remark: it seems wiki pages have a problem. I see HTML code in place of the rendered viewing. A GitHub bug?

Yes, very strange Github issue; I solved it by doing a hard refresh of the page (command+shift+R on my Mac).

> I'm always using the --skip-cleaning option, and it seems to be properly recognized.

Yes, that's the correct way!

> Also about tokenization, does it work for Czech or Slovak?

Yes, our pipeline fully supports both languages and the language codes are correct: cs and sk.
